### ISSUE TITLE
Apply filter to schema.siteUrl

### ISSUE THREAD

**Issue by matepaiva** [ashhitch/wp-graphql-yoast-seo#99]
As graphql can be used for headless websites, sometimes the siteUrl is not the same as the `get_site_url()` result.

If we apply a filter to this value, we open to the possibility to manipulate it right before resolving it to GraphQL. And we don't introduce any breaking change :)

```diff
# plugins/wp-graphql-yoast-seo/wp-graphql-yoast-seo.php:710
- 'siteUrl' => wp_gql_seo_format_string(get_site_url()),
+ 'siteUrl' => wp_gql_seo_format_string(apply_filters('wp_gql_seo_site_url', get_site_url())),
```

Before opening this issue, I tried to `add_filter` directly to `site_url`. Although it works, it also breaks other parts of the WordPress panel that uses this global variable. That's why I am now suggesting this approach.

Do you mind if I open a Pull Request with that change? I would love to contribute to this project.

---

## Versions
```
- "ashhitch/wp-graphql-yoast-seo": "^4.14",
- "wpackagist-plugin/wordpress-seo": "^16.1",
```

**Comment by ashhitch** [comment]
@matepaiva  yes please open a PR for this

**Comment by matepaiva** [comment]
@ashhitch could you take a look at the PR please? :)

**Comment by ashhitch** [comment]
@matepaiva sorry life's been a bit crazy, I have merged this in and will get a release out soon

**Comment by ashhitch** [comment]
merged in v4.14.1

**Comment by matepaiva** [comment]
Thank you, I can see it's already working :)
